src/utils/typography.ts

```diff
@@ -0,0 +1,75 @@
+// This file provides access to the 4 typography helpers from the Guardian Design System (@src-foundations/typography)
+// with the output of each helper extended to include the font faces expected by both pre- and post-design system platforms (i.e. Frontend vs DCR)
+// This ensures we can develop our components unaware of platform differences whilst taking advantage of convenience of the design system helper functions.
+// NOTE - meant to be used as temporary tool to abstract those differences until we find a solution on the platform side.
+import {
+    titlepiece as dsTitlepiece,
+    headline as dsHeadline,
+    body as dsBody,
+    textSans as dsTextSans,
+} from '@guardian/src-foundations/typography';
+
+// Copied over from @src/foundations
+// Defines the FontScaleArgs argument allowed into any typography helper
+type LineHeight = 'tight' | 'regular' | 'loose';
+type FontWeight = 'light' | 'regular' | 'medium' | 'bold';
+type FontScaleArgs = {
+    lineHeight?: LineHeight;
+    fontWeight?: FontWeight;
+    italic?: boolean;
+};
+
+// Describes any of the helper functions we're mapping from @src/foundations
+interface TypographyHelperFunction {
+    [key: string]: Function;
+}
+
+// Defines the possible sizes for each typography object
+// Used to describe the keys allowed for any of those objects
+type TitlepieceType = 'small' | 'medium' | 'large';
+type HeadlineType = 'xxxsmall' | 'xxsmall' | 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
+type BodyType = 'small' | 'medium';
+type TextSansType = 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
+
+// TITLEPIECE
+const titlepieceSizes = ['small', 'medium', 'large'];
+const titlepiece: TypographyHelperFunction = {};
+for (let i = 0; i < titlepieceSizes.length; i++) {
+    const size = titlepieceSizes[i] as TitlepieceType;
+    titlepiece[size] = (fontScaleArgs: FontScaleArgs): string => `${dsTitlepiece[size](
```

Comment:
I wonder if there's an argument for being looser with the types here and passing whatever we receive directly through to the underlying design system function? Possibly even a good use case for rest params. So that if the design system changes/adding additional params these wrappers would continue to Just Work (tm). What do you think?

Reply:
As discussed offline, I've refactored this after realising the Design System is actually exporting a lot of these types that I had previously copied across. It's much neater now. Only one copied/pasted is `FontScaleArgs`, which is not currently exported, although Simon has offered to export this in the next version of `@guardian/src-foundations` so I'll keep an eye on it.

Reply:
I'm in favour of a more strict approach to the `FontScaleArgs` parameter and I still think we should type check it. I agree it creates a bit more coupling, but it will stop us adding any properties not supported by the Design System (or make any typos on the supported names). The `FontScaleArgs` is going to be exported from the Design System very soon as well, so we'll just import it and use as soon as we can.